packages/mdc-shape/README.md

```diff
@@ -0,0 +1,103 @@
+<!--docs:
+title: "Shape"
+layout: detail
+section: components
+excerpt: "Shapes direct attention, identify components, communicate state, and express brand."
+path: /catalog/shape/
+-->
+
+# Shape
+
+<!--<div class="article__asset">
+  <a class="article__asset-link"
+     href="https://material-components-web.appspot.com/shape.html">
+    <img src="{{ site.rootpath }}/images/mdc_web_screenshots/shape.png" width="159" alt="Shape screenshot">
+  </a>
+</div>-->
+
+Shapes direct attention, identify components, communicate state, and express brand.
+
+Many MDC Web components include direct support for rounded corners; this package provides additional support for
+applying angled corners to unelevated surfaces.
+
+## Design
+
+<ul class="icon-list">
+  <li class="icon-list-item icon-list-item--link">
+    <a href="https://material-components-web.appspot.com/shape.html">Demo</a>
+  </li>
+</ul>
+
+## Installation
+
+```
+npm install @material/shape
+```
+
+## Basic Usage
+
+### HTML
+
+The markup for angled corners involves a container element, and one element per angled corner.
+
+The following example demonstrates angled corners applied to a MDC Button, but this technique may be applied to any
+unelevated component.
+
+```html
+<div class="mdc-shape-angled-corner-container my-shape-container">
+  <button class="mdc-button mdc-button--unelevated">Button</button>
+  <div class="mdc-shape-angled-corner-container__corner mdc-shape-angled-corner-container__corner--top-left"></div>
+  <div class="mdc-shape-angled-corner-container__corner mdc-shape-angled-corner-container__corner--top-right"></div>
+  <div class="mdc-shape-angled-corner-container__corner mdc-shape-angled-corner-container__corner--bottom-right"></div>
+  <div class="mdc-shape-angled-corner-container__corner mdc-shape-angled-corner-container__corner--bottom-left"></div>
+</div>
+```
+
+> **Note:** Each angled corner should have its own `<div>` element. Any corner that will not be cut (i.e. its size is 0)
+> does not need a dedicated element.
+
+### Styles
+
+```scss
+@import "@material/shape/mixins";
+
+.my-shape-container {
+  @include mdc-shape-angled-corner(#fff, 10px);
+}
+```
+
+## Variants
+
+### Stroked Angled Corners
+
+Stroked angled corners involve the same markup and styles as above, with the addition of including a mixin for stroke:
+
+```scss
+@import "@material/shape/mixins";
+
+.my-shape-container {
+  @include mdc-shape-angled-corner(#fff, 10px);
+  @include mdc-shape-angled-corner-stroke(2px, blue);
+}
+```
+
+## Style Customization
+
+### CSS Classes
+
+CSS Class | Description
+--- | ---
+`mdc-shape-angled-corner-container` | Mandatory. Parent element containing the component to be masked.
```

Comment:
Depending on whether or not we have other different angled corners, we may want `angled-corner` to be a modifier class: `mdc-shape-conatiner--angled-corner`. The corners could then be:

`mdc-shape-container__corner`
`mdc-shape-container__corner--bottom-left`
`mdc-shape-container__corner--bottom-right`
`mdc-shape-container__corner--top-left`
`mdc-shape-container__corner--top-right`

Reply:
I'm still thinking over this, but it occurs to me that I want a 2nd review for this anyway so I'll leave this for @advorak to chime in on potentially as well.

Reply:
I definitely like @moog16's class names cuz they're a bit shorter and easier to read.

But it sort of depends on whether we think there's going to be other _non-angled_ corner geometries in the future, and how we would want to name them.

So I guess I would go with Matty's suggestion unless we're fairly confident that those names would collide with other corner types in the future.

Reply:
I experimented with changing this, and realized that having a modifier class on the container probably isn't even necessary - there are no OOTB styles that would warrant it, only the styles within the mixins, which would ostensibly be invoked only on selectors for specific cases anyway.